src/projections/Tools/PerformanceCounters/ThreadedFileReader.java

```diff
@@ -56,8 +57,10 @@ public void run() {
 				if (logdata.type == ProjDefs.END_PROCESSING) {
 					int EPid = MainWindow.runObject[myRun].getEntryIndex(logdata.entry);
 					for (int i = 0; i < numPerfCounts; ++i) {
-						perfCounters[i][pIdx][EPid] += logdata.perfCounts[i] - reader.getLastBE().perfCounts[i];
```

Comment:
This change is necessary because the new semantics of `getLastBE` sets the last begin event to invalid (read: `null`) as soon as an `END_PROCESSING` event is read. It previously only set it to invalid when an `END_IDLE` even was hit (now ending events of both types set it to invalid).

Reply:
I think this should become a comment here and / or on line 46 to explain this interaction, since it seems like a non-obvious thing you have to be aware of.